odl/solvers/nonsmooth/douglas_rachford.py

```diff
@@ -160,32 +160,54 @@ def douglas_rachford_pd(x, f, g, L, tau, sigma, niter,
     w1 = x.space.zero()
     w2 = [Li.range.zero() for Li in L]
 
+    # Temporaries (not in original article)
+    tmp_domain = x.space.zero()
+
     for k in range(niter):
-        tmp_1 = sum(Li.adjoint(vi) for Li, vi in zip(L, v))
-        tmp_1.lincomb(1, x, -tau / 2, tmp_1)
-        f.proximal(tau)(tmp_1, out=p1)
+        lam_k = lam(k)
+
+        # Compute tmp_domain = sum(Li.adjoint(vi) for Li, vi in zip(L, v))
+        L[0].adjoint(v[0], out=tmp_domain)
```

Comment:
Do we allow `L` empty? we did not before anyway.

I guess we can add an if statement for it, but I'm not even sure if the algorithm is valid in that case.